packages/service-core/src/sync/sync.ts

```diff
@@ -318,9 +318,18 @@ async function* bucketDataBatch(request: BucketDataRequest): AsyncGenerator<Buck
         // Send the object as is, will most likely be encoded as a BSON document
         send_data = { data: r };
       } else if (raw_data) {
-        // Data is a raw string - we can use the more efficient JSON.stringify.
+        /**
+         * Data is a raw string - we can use the more efficient JSON.stringify.
+         * FIXME: ensure that the data is actually a string
+         */
         const response: util.StreamingSyncData = {
-          data: r
+          data: {
+            ...r,
+            data: r.data.map((entry) => ({
+              ...entry,
+              data: typeof entry.data == 'string' ? entry.data : JSONBig.stringify(entry.data)
+            }))
```

Comment:
Why were changes needed here?

Reply:
I experienced an issue when testing the Postgres bucket storage over an HTTP stream. 

The `OplogEntry` here has a `data` member which is defined of the type ` data?: Record<string, SqliteJsonValue> | string | null;` and `SqliteJsonValue` supports `number | string | bigint | null`. In MongoDB the data is stored internally and returned (at this point) as a serialised JSON string which works fine. For Postgres I've stored the data as a `bytea` column containing serialised BSON assuming this might make blob/binary data storage simpler in future. The Postgres bucket storage deserialises the BSON to a ` Record<string, SqliteJsonValue>` compatible document which satisfies the `OplogEntry::data` typing, but causes issues in the following `JSON.stringify` call since `bigint` cannot be serialized. 

I've added the check here to continue using `JSON.stringify` for the final response. Alternatively we could use `JSONBig.stringify`, but I noticed the comment above mentions it is less efficient.

Reply:
The `OplogEntry` type does make things a little confusing, since it's used for both the bucket storage APIs, and the protocol, but differs a bit between them (currently data is always `string | null` in mongodb storage).

Storing as BSON in Postgres is an interesting idea. I think that would make sense if we start syncing rows directly as BSON in the future. However, since we always sync as a string right now, doing that `JSONBig` serialization for every client adds a _lot_ of overhead. So I'd recommend just storing it as a string for now, and re-evaluate this when we do add proper binary syncing support later.